File: 08-26/adts/bst/BST.java
package adts.bst;

public interface BST<T extends Comparable<T>> extends Iterable<T> {
    public void insert(T data);

    public void remove(T data);

    public boolean contains(T data);

    public T get(T data);

    public T max();

    public T min();

    public int count();

    public void printInOrder();
}

File: 08-26/adts/bst/bst.cpp
#pragma once

#include "../iterator/iterable.cpp"

template <class T>
class bst : iterable<T>
{
public:
    virtual void insert(T data) = 0;
    virtual void remove(T data) = 0;
    virtual bool contains(T data) = 0;
    virtual T get(T data) = 0;
    virtual T max() = 0;
    virtual T min() = 0;
    virtual int count() = 0;
    virtual void printInOrder() = 0;
};
